ImageGallery/ImageGallery/Views/MainScreen.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  MainScreen.swift
+//  ImagaeGallery
+//
+//  Created by Nick Ryan on 6/22/24.
+//
+
+import SwiftUI
+
+struct SearchBar: View {
+  @Binding var searchQuery: String
+  let onSearch: () -> Void
+  
+  var body: some View {
+    HStack {
+      TextField("Search Images...", text: $searchQuery, onCommit: {
+        onSearch()
+      })
+      .textFieldStyle(RoundedBorderTextFieldStyle())
+      .padding()
+      
+      Button(action: {
+        onSearch()
+      }) {
+        Text("Search")
+      }
+    }
+  }
+}
+
+
+
+
+
+struct ImageGrid: View {
+  @ObservedObject var imageStore: ImageStore
+  @State private var isDownloading = false
+  @State private var downloadProgress: Float = 0.0
+  
+  var body: some View {
+    ScrollView {
+      LazyVGrid(columns: [GridItem(.adaptive(minimum: 150))]) {
+        ForEach(imageStore.photos) { photo in
+          NavigationLink(destination: PhotoDetail(photo: photo)) {
+            ImageSquare(photo: photo, isDownloading: $isDownloading,
+                        downloadProgress: $downloadProgress, downloadAction: {
+              Task {
+                await download(photo: photo)
+              }
+            })
+          }
+        }
+      }
+    }
+  }
+  
+  
+  private func download(photo: PexelsPhoto) async {
```

Comment:
Instead of making this here, you should do this in the other view. The state of `isDownloading` and `downloadProgress` is being share between all views. That's why when you tap to download 1 image it shows a loading indicator in all of them.